src/server.c

```diff
@@ -4632,37 +4631,16 @@ int redisSupervisedUpstart(void) {
 
 int redisCommunicateSystemd(const char *sd_notify_msg) {
     const char *notify_socket = getenv("NOTIFY_SOCKET");
```

Comment:
From your comment in the PR, I think I have to agree that the option does not seem to make much sense anymore? I'd personally just call this function unconditionally, and deprecate the config option. If the caller, be that systemd or start-stop-daemon, sets the `NOTIFY_SOCKET` environment variable, then `sd_notify()` will send the notification, otherwise not. This looks like the most common behavior used everywhere else.

But perhaps that should be in another commit, if the change is desired.